### ISSUE TITLE
Inconsistent out structure when no test folder is available

### ISSUE THREAD

**Issue by fcrespo82** [Microsoft/vscode#2557]
When I build a extension with test folder the out folder is structured as (other files/folders removed for clarification)

    out/
        src/
            extension.ts
        test/
            index.ts

but, without the test folder the out folder is structured as

    out/
        extension.ts

Which leads the extension to fail the activation process. Because in the `package.json` there is a directive ("main") that tells that the `extension.js` is in `./out/src/extension`

## Steps to reproduce

In the command line 
```
1. yo code
2. cd into the extension directory
3. Choose New extension (TypeScript)
4. npm run vscode:prepublish 
5. ls out
6. Inspect the output
```
```
1. yo code
2. cd into the extension directory
3. Choose New extension (TypeScript)
4. rm -rf test
5. npm run vscode:prepublish
6. ls out
7. Inspect the output
```

In my opinion it shoud output the structure  below

    out/
        src/
            extension.ts

**Comment by bpasero** [comment]
Wow, I can reproduce and this looks like a bug in TypeScript. @dbaeumer @jrieken any idea why this would happen?

**Comment by dbaeumer** [comment]
Actually, that is intended by TS. They always generated the shortest path to make the file unique. The fix would be on our end by adding a rootDir property to tsconfig.json. I think:

```json
{
	"compilerOptions": {
		"module": "commonjs",
		"target": "es5",
		"outDir": "out",
		"noLib": true,
		"sourceMap": true,
		"rootDir": "."
	},
	"exclude": [
		"node_modules"
	]
}
```

should do the trick

**Comment by dbaeumer** [comment]
@bpasero see my latest comment.

**Comment by bpasero** [comment]
Great, I added this to the generated TS extension. @fcrespo82 please update your tsconfig.json and add "rootDir": ".".

**Comment by fcrespo82** [comment]
Awesome, I was going to open a pull request in Microsoft/vscode-generator-code but I saw it is already included in Microsoft/vscode-generator-code@eb57110b2526f059703389021647b109a5978414
